src/engine/controls/bpmcontrol.cpp

```diff
@@ -164,10 +169,10 @@ void BpmControl::slotFileBpmChanged(double file_bpm) {
     // engine BPM. We only do this for SYNC_NONE decks because EngineSync will
     // set our BPM if the file BPM changes. See SyncControl::fileBpmChanged().
     //qDebug() << "BpmControl::slotFileBpmChanged" << file_bpm;
-    BeatsPointer pBeats = m_pBeats;
```

Comment:
How could the member become invalid while executing a slot?? Then this design is broken somewhere else. I've seen this pattern at numerous places without even the slightest mentioning of why this has been done. And I guess some of those local stack variables have already been eliminated at other occurrences.

Reply:
In case a new track is loaded in BpmControl::trackLoaded called from the engine worker thread. 
The slot is directly connected and are executed via the GUI or the controller thread. 
Now it can become null just after if() and before the deference ->.